baseplate/observers/metrics_tagged.py

```diff
@@ -0,0 +1,197 @@
+from enum import Enum
+from typing import Any
+from typing import Dict
+from typing import List
+from typing import Optional
+
+from baseplate import _ExcInfo
+from baseplate import BaseplateObserver
+from baseplate import LocalSpan
+from baseplate import RequestContext
+from baseplate import Span
+from baseplate import SpanObserver
+from baseplate.lib import config
+from baseplate.lib import metrics
+from baseplate.observers.timeout import ServerTimeout
+
+
+class Errors(Enum):
+    TIMED_OUT = 1
+    EXCEPTION = 2
+
+
+error_messages = {
+    Errors.TIMED_OUT: "timed_out",
+    Errors.EXCEPTION: "internal_server_error",
+}
+
+
+class TaggedMetricsBaseplateObserver(BaseplateObserver):
+    """Metrics collecting observer.
+
+    This observer reports metrics to statsd in the Influx StatsD format. It does three important things:
+
+    * it tracks the time taken in serving each request.
+    * it batches all metrics generated during a request into as few packets
+      as possible.
+    * it adds tags to the metric if they are in the config tag whitelist
+
+    The batch is accessible to your application during requests as the
+    ``metrics`` attribute on the :py:class:`~baseplate.RequestContext`.
+
+    :param client: The client where metrics will be sent.
+    :param cfg: the parsed application config with the tag whitelist
+
+    """
+
+    def __init__(self, client: metrics.Client, whitelist: List[str]):
+        self.client = client
+        self.whitelist = whitelist
```

Comment:
Should we convert `whitelist` to a `set` since we're just checking `x in whitelist`?